### PR TITLE
Rename `Prot` struct, `protection` member, `prot()` to visibility/vis

### PR DESCRIPTION
**Pull request by Geod24**
The way we approach attributes such as `public`, `private`, etc...
was overhauled as part of DMD v2.071.0 (2016-04-05).
Those changes kicked off a long transition period, which is now over,
changing the idea of such attributes being "protection attributes"
to them being visibility (notably fixing issues with `private`
symbols in different modules conflicting).

However, internal compiler structures were never changed to reflect
this meaning, and the name `protection` and its abbreviation `prot`
are still everywhere. This can be misleading for new and seasoned
contributors alike.

This commit mass-rename usages of "protection" & co to "visibility".
As "vis" is not quite an obvious abbreviation, "visible" was used
for the method name.

This finishes the rename started in 34ef34d6522bb134032fc6b00a0d5de1f9b5da77,
however a few items are left on this list:
- Adding a `getVisibility` alias for the `getProtection` trait;
- Adding a `visibility` alias in the DDOC output;
- Some files rename in the test suite;

### CODE REVIEW

File: src/dmd/aggregate.d [hunk 1]
```diff
@@ -125,14 +125,14 @@ extern (C++) abstract class AggregateDeclaration : ScopeDsymbol
 
     Expression getRTInfo;   /// pointer to GC info generated by object.RTInfo(this)
 
-    Prot protection;                /// visibility
+    Visibility visibility;                /// visibility
```
Comment (WalterBright) [inline src/dmd/aggregate.d:128]: The comment should say something illuminating rather than redundant.

File: src/dmd/backend/dcgcv.d [hunk 2]
```diff
@@ -74,7 +74,7 @@ extern (C) void TOOFFSET(void* p, targ_size_t value)
 
 extern __gshared char* ftdbname;
 
-// Convert from SFL protections to CV4 protections
+// Convert from SFL visibilitiess to CV4 protections
```
Comment (WalterBright) [inline src/dmd/backend/dcgcv.d:77]: visibilities

**Comment by WalterBright** [comment]
I'm ok with this sort of renaming only because it's what the spec says, and having the spec congruent with the implementation is a good thing. Just fix the couple things I commented on, and it'll be good to go. Thanks!

**Comment by Geod24** [comment]
Comments addressed:
- Regarding the comment, I think the field name is self explanatory, so I just made it `///` (so it shows up in DDOC);
- Typo fixed

**Review by WalterBright — APPROVED** [review]

**Comment by Geod24** [comment]
Anything blocking this ? There was a spurious `std.parallelism` failure on the auto-tester, I just re-triggered it.

**Pull request merged**